### PR TITLE
Teams better permissions loading

### PR DESCRIPTION
**Pull request by buoyad**
- Use `TeamRoleMap` to generate team permissions
  - Remove `leaveTeam` which depends on extra data and was unused
  - Remove `setMemberShowcase` which depends on extra data
- Remove `canUserPerform` loading logic
- Also: Fix navigating to teams throughout the app

### CODE REVIEW

**Comment by buoyad** [comment]
This is ready cc @keybase/y2ksquad

**Comment by buoyad** [comment]
@keybase/react-hackers bump

**Comment by buoyad** [comment]
@keybase/react-hackers bump 〰️

File: shared/chat/conversation/info-panel/menu/container.tsx [hunk 1]
```diff
@@ -25,6 +26,7 @@ const moreThanOneSubscribedChannel = (
   metaMap: Container.TypedState['chat2']['metaMap'],
   teamname?: string
 ) => {
+  // TODO don't use metamap here, it only has a subset of your convs
```
Comment (chrisnojima) [inline shared/chat/conversation/info-panel/menu/container.tsx]: maybe just chagne to use inboxlayout

Comment (buoyad) [reply shared/chat/conversation/info-panel/menu/container.tsx]: I'll put this in #20871

**Pull request merged**